File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
using namespace std;

int main() {

	int side1;
	int side2;
	int side3;

	while (true)
	{
		cin >> side1 >> side2 >> side3;

		if (side1 == 0 && side2 == 0 && side3 == 0)
		{
			break;
		}
		else {
			if (max(max(side1, side2), side3) >= side1 + side2 || max(max(side1, side2), side3) >= side1 + side3 || max(max(side1, side2), side3) >= side2 + side3)
			{
				cout << "Invalid" << endl;
			}
			else if (side1 == side2 && side2 == side3)
			{
				cout << "Equilateral" << endl;
			}
			else if (side1 == side2 || side1 == side3 || side2 == side3)
			{
				cout << "Isosceles" << endl;
			}
			else {
				cout << "Scalene" << endl;
			}
		}
	}
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        while (true) {
            int side1 = sc.nextInt();
            int side2 = sc.nextInt();
            int side3 = sc.nextInt();
            
            if (side1 == 0 || side2 == 0 || side3 == 0) {
                break;
            } else if (side1 >= side2 + side3 || side2 >= side1 + side3 || side3 >= side1 + side2) {
                System.out.println("Invalid");
            } else if (side1 == side2 && side2 == side3) {
                System.out.println("Equilateral");
            } else if (side1 == side2 || side2 == side3 || side1 == side3) {
                System.out.println("Isosceles");
            } else {
                System.out.println("Scalene");
            }
        }
    }
}